tensorflow_probability/python/distributions/empirical.py

```diff
@@ -0,0 +1,187 @@
+# Copyright 2018 The TensorFlow Probability Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ============================================================================
+"""The Empirical distribution class."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import tensorflow as tf
+from tensorflow.python.ops import array_ops, gen_array_ops
```

Comment:
array_ops shouldn't be needed.
array_ops.reshape -> tf.reshape
array_ops.shape -> tf.shape

do we need to use unique_v2, or can we use the one in the TF API (tf.unique[_with_counts])?

Reply:
Hi @brianwa84 , I removed `array_ops` in the latest PR.

I use `unique_v2` because it supports `axis` for multidimensional tensors.
It seems that Tensorflow is planning to port this feature to TF API, but no progress has been made in the latest release yet.(https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/ops/array_ops.py#L1263)

Reply:
The PR to switch to v2 seems to stall for some reason https://github.com/tensorflow/tensorflow/pull/17282